test/template/www/js/scenarioSyncRestart30.js

```diff
@@ -0,0 +1,47 @@
+var app = {
+    // Application Constructor
+    initialize: function () {
+        this.bindEvents();
+    },
+    bindEvents: function () {
+        document.addEventListener('deviceready', this.onDeviceReady, false);
+    },
+    onDeviceReady: function () {
+        app.receivedDeviceReady();
+    },
+    // Update DOM on a Received Event
+    receivedDeviceReady: function () {
+        document.getElementById("deviceready").innerText = "Device is ready (scenario - sync)";
+        console.log('Received Event: deviceready');
+        /* invoke sync with no UI options */
+        window.codePush.sync(
+            function (status) {
+                // only output result statuses
+                switch(status) {
+                    case SyncStatus.UP_TO_DATE:
+                    case SyncStatus.UPDATE_INSTALLED:
+                    case SyncStatus.UPDATE_IGNORED:
+                    case SyncStatus.ERROR:
+                    case SyncStatus.IN_PROGRESS:
+                        app.sendTestMessage("SYNC_STATUS", [status]);
+                    default:
+                        break;
+                }
+            },
+            {
+                installMode: InstallMode.ON_NEXT_RESTART,
+                minimumBackgroundDuration: 30
```

Comment:
You can use the VS emulator for android on Windows, which is a lot faster. I don't remember if I used it for running the tests, you might need to change the [server url](https://github.com/Microsoft/cordova-plugin-code-push/blob/master/gulpfile.js#L145) to your machine's [localhost loopback](http://stackoverflow.com/questions/28262276/visual-studio-2015-ctp5-android-emulator-loopback-to-localhost/32660110#32660110)